src/generator/iter.rs

```diff
@@ -193,6 +194,27 @@ impl<'a> Iter<'a> {
                 }))
             },
             Include::Pdf(path) => Ok(Event::Pdf(Pdf { path })),
+            Include::Graphviz(path) => {
+                let content = fs::read_to_string(&path).map_err(|err| {
+                    gen.diagnostics()
+                        .error("can't read graphviz file")
+                        .with_error_section(range, "in this include")
+                        .error(format!("cause: {}", err))
+                        .note(format!("reading from path {}", path.display()))
+                        .emit();
+                    Error::Diagnostic
+                })?;
+                let tag = Tag::Graphviz(Graphviz {
+                    label,
+                    caption,
+                    scale,
+                    width,
+                    height,
+                });
+                self.peek.push_back((WithRange(Event::Text(content.into()), range), self.last_kind));
+                self.peek.push_back((WithRange(Event::End(tag.clone()), range), self.last_kind));
```

Comment:
Heh. Cute hack. It's not even proper to really say this is a hack. Does this conflict or overlap with the ability to do multiple peeks?

Reply:
It could enable multiple peeks. For now, the `peek` method still only peeks one element.

Reply:
I recall some instance where multiple peeks (using itertools) was already done. Was that in another branch or am I imagining things?

Reply:
So no conflict. Okay.